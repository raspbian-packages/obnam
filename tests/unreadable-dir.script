#!/bin/sh
# Copyright 2011  Lars Wirzenius
# 
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.

set -e

echo aaa > "$DATADIR/data/aaa"
mkdir --mode=0 "$DATADIR/data/bbb"
echo ccc > "$DATADIR/data/ccc"

if $SRCDIR/tests/backup
then
    exit=0
else
    exit=1
fi
$SRCDIR/tests/restore

# Remove the problematic directory so that verify works.
# Don't do this if running as root, since in that case
# obnam _can_ back it up. (Yes, this is convoluted.)
# When removing the directory, make sure the mtime doesn't
# change of the parent.
if [ "$(whoami)" != root ]
then
    touch -r "$DATADIR/data" "$DATADIR/timestamp"
    rmdir "$DATADIR/data/bbb"
    touch -r "$DATADIR/timestamp" "$DATADIR/data"
fi

$SRCDIR/tests/verify

exit $exit

