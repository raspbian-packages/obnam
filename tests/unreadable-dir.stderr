ERROR: There were errors during the backup
